### PR TITLE
Add dupin demo

### PR DESCRIPTION
**Pull request by MarcosDY**
Add dupin demo, with spiffe-nginx integration

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,6 +24,10 @@ Demonstrates two workloads communicating via mutually authenticated Envoys using
 
 Demonstrates 100 workloads on 100 servers managed by one spire-server
 
+**[dupin](dupin) - nginx with SPIFFE support**
+
+Demonstrates the use of the SPIFFE Workload API to automatically get X.509 certificates in nginx. Connections are accepted or rejected based on allowed SPIFFE IDs.
```
Comment (drrt) [inline README.md]: Maybe say "natively with nginx, with no helper"

File: dupin/spiffe-nginx/nginx/vendor.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+		make install
+		cd third_party/protobuf
+		make install
+	)
+}
+
+mkdir -p ${VENDOR_DIR}
+do_pcre
+do_grpc
+rm -rf ${VENDOR_DIR}
```
Comment (drrt) [inline dupin/spiffe-nginx/nginx/vendor.sh]: Is this file necessary if the demo is not building nginx?

Comment (MarcosDY) [reply dupin/spiffe-nginx/nginx/vendor.sh]: yes, Its installing C code than modules need, without this instalations we got errors because some dependencies are no found
